src/ophyd_async/epics/areadetector/areadetector.py

```diff
@@ -0,0 +1,45 @@
+from typing import Sequence, Tuple, cast
+
+from ophyd_async.core import DirectoryProvider, SignalR, StandardDetector
+
+from .controllers import ADController
+from .drivers import ADDriver, ADDriverShapeProvider
+from .writers import HDFWriter, NDFileHDF
+
+
+class AreaDetector(StandardDetector):
+    _controller: ADController
+    _writer: HDFWriter
+
+    def __init__(
+        self,
+        prefix: str,
+        directory_provider: DirectoryProvider,
+        name: str = "",
+        config_sigs: Sequence[Tuple[str, SignalR]] = (),
+    ):
+        self.drv = ADDriver(prefix + "DRV:")
+        self.hdf = NDFileHDF(prefix + "HDF:")
+
+        super().__init__(
+            ADController(self.drv),
+            HDFWriter(
+                self.hdf,
+                directory_provider,
+                lambda: self.name,
+                ADDriverShapeProvider(self.drv),
+            ),
+            config_sigs=config_sigs,
+            name=name,
+        )
+
+    async def connect(self, sim: bool = False):
+        await super().connect(sim=sim)
+        driver = self._controller.driver
+        self._config_sigs = [
+            *self._config_sigs,
+            *[
+                (getattr(signal, "name"), cast(SignalR, signal))
+                for signal in [driver.acquire_time, driver.acquire]
+            ],
+        ]
```

Comment:
What does this do?

Reply:
I guess this is something to do with `config_sigs` becoming `Sequence[Tuple[name, SignalR]]`. Why did it change from `Sequence[SignalR]`? I was envisioning that `config_sigs` would be named and connected as per its position in the tree, and so `StandardDetector` didn't have to do anything apart from read them.

E,g.
```python
class MyDet(StandardDetector):
    def __init__(...):
        self.drv = MyDrv(...)
        ...
        super().__init__(
            config_sigs = [self.drv.acquire_time, self.drv.gain]
        )
```
Now if we name our `MyDet` instance `"my_det"` then we would get drv and all its children connected in `connect()`, named in `set_name()`, and `my_det-drv-acquire_time` and `my_det-drv-gain` would appear as names in `read()`.

Reply:
config_sigs could be any signals; they could be signals associated with the detector, or not. For this reason, when doing something like:

```
with DeviceCollector():
    my_detector = Detector(..., config_sigs=[signal_a, signal_b])
```

I realised that if signal_a and b hadn't been connected and named properly, there would be problems with the bluesky bundler when doing things like opening a run. So, I added a way to ensure they were connected and named properly; by passing these config signals as a tuple of the name you'd like to appear on a descriptor doc, and the signal, the StandardDetector class ensures they are named and connected properly.

Perhaps it's easier to discuss this one offline; but the way I see it we have two approaches to this problem:
1. constrain config_sigs to just be stuff that the detector knows about.
2. raise an error somewhere (in bluesky, maybe?) if we try to use config sigs that aren't named, and another error if we try to use config_sigs that haven't been connected.
3. do what I've done above; allow any config sigs as long as I also get a tuple with a name, so I know what to name them. Because the DeviceCollector gets that from a context.

Thinking about it, number 2 might be best. Effectively the error I get from the bundler boils down to the fact that, if the signals aren't named, we end up with a dictionary for config sigs with just one key ("") and something really doesn't like it being empty. Or, I think I want to make a test case for this, fix it, make a PR with bluesky but still keep what I've done above. Would like to know your thoughts on it.

Reply:
For the moment, I've added a function that, each time you stage, checks that the config signals are connected and named.

But doing what you suggests sound like a good idea also.